src/components/CGroupHeader.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <tr @click="onToggle" class="text-xs hover:bg-blue-100">
+    <td class="border px-4 py-2">
+      <div class="flex">
+        <img v-if="isOpened" src="@/assets/img/expand_more.svg" width="20rem" />
+        <img v-else src="@/assets/img/expand_less.svg" width="20rem" />
+        <pre class="text-xs ml-2 whitespace-normal">{{ groupId }}</pre>
+      </div>
+    </td>
+    <td class="border px-4 py-2">
+      {{ remainingTime }}
+    </td>
+    <td class="border px-4 py-2">
+      {{ progress | percentage }}
+    </td>
+    <td class="border px-4 py-2">
+      {{ messageCount }}
+    </td>
+    <td class="border px-4 py-2">
+      <ul>
+        <li :key="index" v-for="(actor, index) in uniqueActors">
+          {{ actor }}
+        </li>
+      </ul>
+    </td>
+  </tr>
+</template>
+<script>
+import utils from '@/utils';
+
+export default {
+  name: 'CGroupHeader',
+  props: {
+    messages: Array,
+    groupId: String
+  },
+  data() {
+    return {
+      isOpened: false
+    };
+  },
+  computed: {
+    messageCount() {
+      return this.messages.length;
+    },
+    uniqueActors() {
+      return new Set([...this.messages].map(item => item.actorName));
```

Comment:
no need for `[...this.messages]` here, as you are constrction a new array with map.